src/phylum/console.py

```diff
@@ -0,0 +1,46 @@
+"""Provide a console object that can be used throughout the application.
+
+The `rich` library is used here for excellent control over the console.
+Reference: https://rich.readthedocs.io/en/latest/index.html
+"""
+
+from rich.console import Console
+from rich.theme import Theme
+
+# Create a custom Phylum theme by setting unique style values for a subset of the markdown styles that `rich` supports.
+# The full list of markdown styles, as of the time of this writing, is included here to make it easier to understand
+# what can be modified. The default styles are also included so any future changes will not cause unexpected output.
+# The colors chosen are from the standard 8-bit colors supported in terminals and are close to the range of blue colors
+# that make up the Phylum logo. These colors are also able to be represented by a name instead of a hex or RGB value.
+# References:
+#   * https://rich.readthedocs.io/en/latest/style.html
+#   * https://rich.readthedocs.io/en/latest/appendix/colors.html
+PHYLUM_STYLES = {
+    # Styles that have been modified from the default:
+    "markdown.code": "bold reverse",
+    "markdown.h1.border": "dodger_blue2",
+    "markdown.h1": "dodger_blue2",
+    "markdown.h2": "deep_sky_blue3",
+    "markdown.h3": "underline cornflower_blue",
+    "markdown.h4": "steel_blue1",
+    "markdown.h5": "turquoise2",
+    "markdown.h6": "cyan1",
```

Comment:
Are these using RGBA colors? There's already 4 different blues in the actual terminal colorscheme, isn't that enough?

I know this isn't exactly about CLI usage, but I personally much prefer when terminal output doesn't rely on RGBA output.

Reply:
My understanding is that these colors are all RGB and do _not_ contain an alpha channel. They all come from the [this page](https://rich.readthedocs.io/en/latest/appendix/colors.html), which makes it sound like they are just english names for each of the 256 colors in the standard 8-bit colors supported in terminals.

Reply:
Oh yeah I didn't mean RGBA, I meant RGB, sorry.

The 88/256/257 colors are not RGB, but there's really no reason to use them over RGB since their purpose overlaps basically.

So my original comment still stands, since the fundamental issue is ultimately just that this uses custom colors rather than the primary/highlighted palette when I don't think there's any reason to. Though as mentioned in CI it doesn't really matter.

Reply:
The choice of colors came from the `rich` package's [documentation on "Styles"](https://rich.readthedocs.io/en/latest/style.html), which indicates specifying colors from their listing of the ["Standard Colors"](https://rich.readthedocs.io/en/latest/appendix/colors.html). The names there appear to coincide with a standard color palette, perhaps the [Extended Web Colors](https://en.wikipedia.org/wiki/Web_colors#Extended_colors) or maybe the [X11 Color Names](https://en.wikipedia.org/wiki/X11_color_names)...not sure. In any case, the choice of colors is meant to be fun and coincide with the Phylum logo and "Rich will attempt to pick the closest color it can if your color isn’t available."

Yes...the color and formatting appears to get stripped out when printed in CI environments...so the fun stops there 😜

Reply:
> Rich will attempt to pick the closest color it can if your color isn’t available.

The only thing it can do is check if it has 88, 256, or truecolor and choose based on that. It has zero knowledge of what the colors hex values will actually be when printed by the terminal.

There is zero guarantee that the 88/255 colors are actually what the "standard colors" suggest they might be. There is no such thing as "correct" RGB values for non-truecolor RGB colors. Technically even RGB colors are still not guaranteed to be displayed the same way due to color spaces.

A terminal using 88 colors (URxvt iirc), for example has a different colormap than one with 256 or 257 colors.

I'd hope it just uses RGB when available, since that at least has *some* guarantee that the color is somewhat like what it actually wants. But when rendering to a terminal you should allow the user to pick which colors they actually want, which helps users with visual disabilities for example (and just users that have very specific preferences).

Either way long story short, when possible in a terminal you should **always** use the 16 colors, rather than relying on the extended color formats like 88/256/truecolor. I don't think anyone cares about this in phylum-ci, but I'd hope you might still find this information useful in the future.

Reply:
I do find the information useful...a lot of it is new to me...thanks!